### PR TITLE
DEV: Remove legacy user menu

### PR DESCRIPTION
**Pull request by keegangeorge**
This PR removes the legacy user menu and all its related code.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/widgets/hamburger-menu.js [hunk 1]
```diff
@@ -355,11 +355,7 @@ export default createWidget("hamburger-menu", {
...
 
   html(attrs, state) {
-    if (
-      this.currentUser &&
-      !this.currentUser.redesigned_user_menu_enabled &&
-      !state.loaded
-    ) {
+    if (this.currentUser && !state.loaded) {
       this.refreshReviewableCount(state);
     }
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/widgets/hamburger-menu.js:360]:
```suggestion
```
The condition was `!redesigned_user_menu_enabled`, so I think we can drop this?

File: app/assets/javascripts/discourse/tests/acceptance/do-not-disturb-test.js [hunk 2]
```diff
@@ -112,8 +112,7 @@ acceptance("Do not disturb", function (needs) {
   });
 });
 
-acceptance("Do not disturb - new user menu", function (needs) {
-  needs.user({ redesigned_user_menu_enabled: true });
+acceptance("Do not disturb - user menu", function (needs) {
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/tests/acceptance/do-not-disturb-test.js:115]: I suspect we'll need to update some of the other tests in this file so that they work with the new user menu. (or, if they're now duplicated, we can delete them)

File: app/controllers/notifications_controller.rb [hunk 3]
```diff
@@ -30,14 +30,10 @@ def index
...
       include_reviewables = false
 
-      if SiteSetting.legacy_navigation_menu?
-        notifications = Notification.recent_report(current_user, limit, notification_types)
-      else
-        notifications =
-          Notification.prioritized_list(current_user, count: limit, types: notification_types)
-        # notification_types is blank for the "all notifications" user menu tab
-        include_reviewables = notification_types.blank? && guardian.can_see_review_queue?
-      end
```
Comment (davidtaylorhq) [inline app/controllers/notifications_controller.rb:40]: This condition is `legacy_navigation_menu`, not the notifications menu. So I don't think we want this change?

File: app/models/notification.rb [hunk 4]
```diff
@@ -279,71 +279,6 @@ def self.prioritized_list(user, count: 30, types: [])
     notifications.to_a
   end
 
-  # TODO(osama): deprecate this method when redesigned_user_menu_enabled is removed
```
Comment (davidtaylorhq) [inline app/models/notification.rb:282]: I think this comment is out-of-date. We're still using this `recent_report` method for the legacy hamburger menu

https://github.com/discourse/discourse/blob/83f1a13374bc3906919744f48071511c12912df9/app/controllers/notifications_controller.rb#L33-L34

File: app/models/user.rb [hunk 5]
```diff
@@ -760,7 +760,7 @@ def total_unread_notifications
   end
 
   def reviewable_count
-    Reviewable.list_for(self, include_claimed_by_others: !redesigned_user_menu_enabled?).count
+    Reviewable.list_for(self).count
```
Comment (davidtaylorhq) [inline app/models/user.rb:763]: The default for `include_claimed_by_others` is true, so I think we need to hard-code false here to keep the correct behaviour ?
```suggestion
    Reviewable.list_for(self, include_claimed_by_others: false).count
```

File: config/locales/server.en.yml [hunk 6]
```diff
@@ -652,8 +652,8 @@ en:
           attributes:
             value:
               invalid_interpolation_keys:
-                one: 'The following interpolation key is invalid: %{keys}'
-                other: 'The following interpolation keys are invalid: %{keys}'
+                one: "The following interpolation key is invalid: %{keys}"
+                other: "The following interpolation keys are invalid: %{keys}"
```
Comment (davidtaylorhq) [inline config/locales/server.en.yml:656]: ? Accidental diff from a merge/rebase?

File: config/locales/server.en.yml [hunk 7]
```diff
@@ -2414,9 +2414,8 @@ en:
...
     default_sidebar_categories: "Selected categories will be displayed under Sidebar's Categories section by default."
     default_sidebar_tags: "Selected tags will be displayed under Sidebar's Tags section by default."
-    enable_new_notifications_menu: "Enables the new notifications menu. Disabling this setting is deprecated. The new notifications menu is always used for non-legacy 'navigation menu' choices. <a href='https://meta.discourse.org/t/260358'>Learn more</a>"
     enable_experimental_hashtag_autocomplete: "EXPERIMENTAL: Use the new #hashtag autocompletion system for categories and tags that renders the selected item differently and has improved search"
-    experimental_new_new_view_groups: "EXPERIMENTAL: Enable a new topics list that combines unread and new topics and make the \"Everything\" link in the sidebar link to it."
+    experimental_new_new_view_groups: 'EXPERIMENTAL: Enable a new topics list that combines unread and new topics and make the "Everything" link in the sidebar link to it.'
```
Comment (davidtaylorhq) [inline config/locales/server.en.yml:2418]: ? Accidental diff from a merge/rebase?

File: config/site_settings.yml [hunk 8]
```diff
@@ -2126,8 +2126,13 @@ navigation:
     choices:
       - "default"
       - "unread_new"
-  enable_new_notifications_menu:
-    default: true
+  enable_custom_sidebar_sections:
```
Comment (davidtaylorhq) [inline config/site_settings.yml:2129]: ? `enable_custom_sidebar_sections` doesn't seem relevant to this PR - accidental diff from a merge/rebase?

File: spec/models/notification_spec.rb [hunk 9]
```diff
@@ -630,137 +630,6 @@ def create(**opts)
     end
   end
 
-  describe "#recent_report" do
```
Comment (davidtaylorhq) [inline spec/models/notification_spec.rb:633]: Per the comment above, I think we still need this `#recent_report` method, so let's restore the test as well.

File: spec/requests/notifications_controller_spec.rb [hunk 10]
```diff
@@ -172,24 +172,6 @@ def delete_notification(resp_code, matcher)
             )
           end
 
-          it "gets notifications list with unread high priority notifications at the top when navigation menu is legacy" do
```
Comment (davidtaylorhq) [inline spec/requests/notifications_controller_spec.rb:175]: This test relates to the navigation_menu, not the notifications menu. So I don't think we should remove it as part of this PR?

File: spec/requests/notifications_controller_spec.rb [hunk 11]
```diff
@@ -286,16 +268,6 @@ def delete_notification(resp_code, matcher)
             )
           end
 
-          it "doesn't include reviewables when navigation menu is legacy" do
-            SiteSetting.navigation_menu = "legacy"
```
Comment (davidtaylorhq) [inline spec/requests/notifications_controller_spec.rb:290]: Likewise, this seems to relate to the navigation menu. Should this diff be part of this PR?

File: spec/serializers/current_user_serializer_spec.rb [hunk 12]
```diff
@@ -283,11 +283,10 @@
       )
     end
 
-    it "isn't included when navigation menu is legacy with old user menu" do
+    it "isn't included when navigation menu is legacy" do
       SiteSetting.navigation_menu = "legacy"
-      SiteSetting.enable_new_notifications_menu = false
 
-      expect(serializer.as_json[:new_personal_messages_notifications_count]).to be_nil
+      expect(serializer.as_json[:new_personal_messages_notifications_count]).to eq(1)
```
Comment (davidtaylorhq) [inline spec/serializers/current_user_serializer_spec.rb:289]: The description of the test says "isn't included", but the expectation is `eq(1)`. We should work out what the desired behaviour is and make things consistent.

Comment (keegangeorge) [reply spec/serializers/current_user_serializer_spec.rb:289]: I think the previous behavior was if the `navigation_menu` was enabled, `enable_new_notifications_menu` would be false?

So I think now that we can have the new notifications menu with the legacy navigation menu, we will need to include the `new_personal_message_notification_count`. I'll update the test description.

**Comment by davidtaylorhq** [comment]
I had a quick look at some of these failures:

- `edit-notification-click-test.js` - seems like a legit failure. Clicking edit notifications in the new user menu does not open the revision history dialog. Let's `skip()` the test for the purposes of this PR, and open a topic on Dev to resolve

- `acceptance/notifications-test.js:16` I'm pretty sure all this ordering logic has been overhauled as part of the new menu. Let's delete this test - it was testing old-menu-specific behaviour

- `sidebar-user-community-section-test.js:981` - I think the message-bus channel in the test needs to be updated to `/reviewable_counts/${loggedInUser().id}`

- `acceptance/user-tips-test.js:15` - `unread_high_priority_notifications` needs to be changed to `new_personal_messages_notifications_count` 

- `unit/routes/review-index-test.js:8` looks like an old-menu-specific test (I think the title has a typo - it's supposed to read "...when <ins>new</ins> user menu not enabled". Let's delete this test

**Review by davidtaylorhq — APPROVED** [review]

**Review by davidtaylorhq — APPROVED** [review]
